runhouse/resources/hardware/cluster.py

```diff
@@ -1128,6 +1128,24 @@ def _start_or_restart_helper(
         # Update the cluster config on the cluster
         self.save_config_to_cluster()
 
+        # Save the local ~/.rh config to the cluster with the user's hashed token,
+        # if such does not exist on the cluster
+        if rns_client.token:
+            if (
+                self.run(
+                    ["[ -f ~/.rh/config.yaml ]"],
+                    node=self.head_ip,
+                    require_outputs=False,
+                )[0]
+                == 0
+            ):
+                logger.debug("Did not change config.yaml")
+            else:
+                source_path = Path("~/.rh/config.yaml").expanduser()
```

Comment:
@jlewitt1 If we want to rsync a condensed config.yaml, it looks like we will have to save it in a temp file (which can be removed after the rsync). 

Pros:
* we save only the necessary fields in the config.yaml we save on the cluster
Cons:
* need to mess a little bit with the user's file system each time he launches a cluster (create a temp file for the condensed config -> rsync it -> delete the temp file+folder)

any thoughts?

Reply:
prob ok to put this in a try / finally block where we delete the temp file after the rsync command is run